### PR TITLE
run module assembly before user manifests

### PR DESCRIPTION
**Pull request by Cosaquee**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/89" target="_blank">ReviewNinja

**Comment by Cosaquee** [comment]
@coi-gov-pl/jboss-puppet-team please review 💃

**Comment by therek** [comment]
:+1:

**Comment by frodoslaw** [comment]
:+1:

File: .gitignore [hunk 1]
```diff
@@ -14,3 +13,5 @@ Vagrantfile
 .DS_Store
 Gemfile.local
 .idea/
+Gemfile.lock
+Gemfile18.lock
```
Comment (cardil) [inline .gitignore]: !resolve why it's ignored. Should be committed as we were talking earlier

Comment (Cosaquee) [inline .gitignore]: Travis can run builds with provided Gemfile, no need to make some more magic in commands. We have Gemfile18 that contains dependencies for Ruby 1.8 so there is no need for Gemfile.lock

File: manifests/user.pp [hunk 2]
```diff
@@ -114,4 +127,8 @@
     }
   }
 
+  anchor { 'jboss::user::before':
```
Comment (cardil) [inline manifests/user.pp]: !fix You couldn't use constant name of the resource in defines. Resource duplication error when managing 2 or more users

File: .gitignore (hunk 1 above)
Comment (Cosaquee) [inline .gitignore]: !resolved

File: manifests/user.pp (hunk 2 above)
Comment (Cosaquee) [inline manifests/user.pp]: !resolved

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -38,6 +38,8 @@ Install the dependencies like so... (you can also pass `--path /fs/path/for/deps
 
     bundle install
 
+If you want to use Ruby 1.8 that we still support you have to pass `--gemfile Gemfile18` to download correct versions of gems that we use.
```
Comment (cardil) [inline CONTRIBUTING.md]: !fix Change to `--gemfile gemfiles/Gemfile18.facter17`

File: Gemfile [hunk 4]
```diff
@@ -11,50 +11,39 @@ group :test do
   gem 'puppetlabs_spec_helper',         :require => false
   gem 'puppet-lint',                    :require => false
   gem 'metadata-json-lint',             :require => false
-  gem 'json',                           :require => false
+  # This package drops support for 1.8
+  gem 'json', '1.8.3',                  :require => false
```
Comment (cardil) [inline Gemfile]: !fix unpin those versions

**Comment by cardil** [comment]
LGTM :+1

File: Gemfile (hunk 4 above)
Comment (Cosaquee) [inline Gemfile]: !resolved

File: CONTRIBUTING.md (hunk 3 above)
Comment (Cosaquee) [inline CONTRIBUTING.md]: !fixed

**Pull request merged**